Bio/Blast/Applications.py

```diff
@@ -1331,13 +1338,21 @@ def __init__(self, cmd="makeblastdb", **kwargs):
                     filename=True,
                     equate=False),
         ]
-        _NcbibaseblastCommandline.__init__(self, cmd, **kwargs)
+        AbstractCommandline.__init__(self, cmd, **kwargs)
+
+    def _input_type_checker(command, x):
+        expression = (x == 'asn1_bin' or
+                      x == 'asn1_txt' or
+                      x == 'blastdb' or
+                      x == 'fasta')
+        return expression
```

Comment:
I see you've only moved this old code, but I think we should simplify it to just:

```python
return x in ('asn1_bin', 'asn1_txt', 'blastdb', 'fasta')
```

Reply:
This is much simpler! I like the change.